### ISSUE TITLE
undefined method `amount' for #<Stripe::SetupIntent

### ISSUE THREAD

**Issue by henrydjacob** [pay-rails/pay#182]
I am facing the same issue "undefined method `amount' for #<Stripe::SetupIntent" where my plan has "trail", any solution on how to handle this?

It fails at /app/views/pay/payments/show.html.erb where line #34

**Comment by henrydjacob** [comment]
Added a conditional display of amount only in case if it is payment_intent? now I am getting "No such payment_intent"

**Comment by excid3** [comment]
Yeah, SetupIntents are for future payments and therefore do not have an amount associated with them.

I think a simple check `if payment.payment_intent?` would work. We'll also want to change the button text from "Pay $20" to "Confirm" or something else.

**Comment by excid3** [comment]
Actually, a lot of the text on this page is specific to payment intents vs setup intents. Might need to change more than just the button.

**Comment by henrydjacob** [comment]
Looks like setup intent is created, calls to `POST /v1/customers/cus_xxxxxxxx/subscriptions` is successful.  It returns the setup_intent

```
  "pending_setup_intent": {
    "id": "seti_1H1rF8FNsQ3YW0Akp",
```

However, when it redirected to `pay/payments/seti_1H1rF8FNsQ3YW0Akp`

I am getting "No such payment_intent: seti_1H1rF8FNsQ3YW0Akp"

The calls to `https://api.stripe.com/v1/payment_intents/seti_1H1rF8FNsQ3YW0Akp/confirm` fail with 404

Should we call a different API for setup intents?

**Comment by excid3** [comment]
I always run the SetupIntent immediately in the Javascript, so I'm not using the Payment page for SetupIntents. The payment page was really only designed for PaymentIntents.

I'd personally recommend updating your Javascript to handle the setup intent's extra confirmation immediately rather than redirecting to this page server side.
